guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,701 @@
+
+
+"""
+Let's look at how our model performs after the fine tuning:
+"""
+
+predictions = model.predict(np.expand_dims(image, axis=0))
+
+classes = {0: "cat", 1: "dog"}
+print("Top class is:", classes[predictions[0].argmax()])
+
+"""
+Awesome - looks like the model correctly classified the image.
+"""
+
+"""
+# Train a Classifier from Scratch
+
+![](https://storage.googleapis.com/keras-nlp/getting_started_guide/prof_keras_advanced.png)
+
+Now that we've gotten our hands dirty with classification, let's take on one
+last task: training a classification model from scratch!
+A standard benchmark for image classification is the ImageNet dataset, however
+due to licensing constraints we will use the CalTech 101 image classification
+dataset in this tutorial.
+While we use the simpler CalTech 101 dataset in this guide, the same training
+template may be used on ImageNet to achieve state of the art scores.
+
+Let's start out by tackling data loading:
+"""
+NUM_CLASSES = 101
+# Change epochs to 100~ to fully train.
+EPOCHS = 1
+
+
+def package_inputs(image, label):
+    return {"images": image, "labels": tf.one_hot(label, NUM_CLASSES)}
+
+
+train_ds, eval_ds = tfds.load(
+    "caltech101", split=["train", "test"], as_supervised="true"
+)
+train_ds = train_ds.map(package_inputs, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(package_inputs, num_parallel_calls=tf.data.AUTOTUNE)
+
+train_ds = train_ds.shuffle(BATCH_SIZE * 16)
+
+"""
+The CalTech101 dataset has different sizes for every image, so we use the
+`ragged_batch()` API to batch them together while maintaining each individual
+image's shape information.
+"""
+
+train_ds = train_ds.ragged_batch(BATCH_SIZE)
+eval_ds = eval_ds.ragged_batch(BATCH_SIZE)
+
+batch = next(iter(train_ds.take(1)))
+image_batch = batch["images"]
+label_batch = batch["labels"]
+
+keras_cv.visualization.plot_image_gallery(
+    image_batch.to_tensor(),
+    rows=3,
+    cols=3,
+    value_range=(0, 255),
+    show=True,
+)
+"""
+
+## Data Augmentation
+
+In our previous finetuning exmaple, we performed a static resizing operation and
+did not include any image augmentation.
+This is because a single pass over the training set was sufficient to achieve
+decent results.
+When training to solve a more difficult task, you'll want to include data
+augmentation in your data pipeline.
+
+Data augmentation is a technique to make your model robust to changes in input
+data such as lighting, cropping, and orientation.
+KerasCV includes some of the most useful augmentations in the keras_cv.layers
+API.
+Creating an optimal pipeline of augmentations is an art, but in this section of
+the guide we'll offer some tips on best practices for classification.
+
+One caveat to be aware of with image data augmentation is that you must be careful
+to not shift your augmented data distribution too far from the original data
+distribution.
+The goal is to introduce noise to prevent overfitting and increase generalization,
```

Comment:
We're using "noise" in two completely different contexts here. Maybe:

The goal is to prevent overfitting and increase generalization, but samples that lie completely out of the data distribution simply add noise to the training process.